app/src/main/java/com/fivegmag/a5gmsmediastreamhandler/MediaSessionHandlerAdapter.kt

```diff
@@ -198,6 +200,14 @@ class MediaSessionHandlerAdapter() {
 
     fun updatePlaybackState(state: String) {
         if (!bound) return
+
+        // trigger reportConsumption when Start/stop of consumption of a downlink streaming session
+        if (PlayerStates.PLAYING == state || PlayerStates.ENDED == state )
+        {
+            println("shilin====reportConsumption triggered by play status[${state}]")
```

Comment:
Check if this is triggered if the user selects a new stream from the dropdown before the current video is finished

Reply:
I checked it's triggered when selects a new stream, but now this piece of code is moved to MSH now.